core/trino-main/src/main/java/io/trino/server/remotetask/TaskInfoFetcher.java

```diff
@@ -228,6 +228,7 @@ private synchronized void sendNextRequest()
 
         errorTracker.startRequest();
         future = httpClient.executeAsync(request, createFullJsonResponseHandler(taskInfoCodec));
+        lastUpdateNanos.set(System.nanoTime());
```

Comment:
`lastUpdateNanos` is supposed to be set after the request completes and we have the response, not when we're sending a new request. We already enforce the limit that only one request can be in-flight at a time inside of `sendNextRequest()`, but if you wanted to reduce the time spent detecting that there's already a request in-flight I see two calls to `getTaskInfo()` at the beginning of `sendNextRequest()` that could reuse the same returned value.

Reply:
Callbacks are executed after future is marked as done and `lastUpdateNanos` is set within the callback. When a coordinator has high load, this could cause more request to be sent as callback could not be executed before 100ms fixed delay.

Reply:
Ah, I see that... maybe we should assign `future = null` in the callback after `lastUpdateNanos` is set and only send another request if `future == null` without checking `future.isDone()` to prevent that? Setting `lastUpdateNanos` before the request completes could still allow another request to be issued before the callback completes in a similar fashion if the response takes more than 100ms to return.

Reply:
Setting this future to null in callback would require to add synchronization/AtomicReference.

Reply:
`updateIntervalMillis` is 3 seconds, but `scheduleWithFixedDelay` is evaluated at 100ms constant rate. Setting `lastUpdateNanos` before send request extends interval for callback evaluation to value of `updateIntervalMillis`. This should reduce possibility to have more request in flight.

Reply:
Repurposing `lastUpdateNanos` to throttle back requests in this way makes the code a bit confusing to read. The intent could easily be missed in future changes.
I'm not sure why we have `updateScheduledExecutor` blindly trying every 100ms. We could mimic ContinuousTaskStatusFetcher and DynamicFiltersFetcher and schedule the next request during callback processing with a delay of `updateIntervalMillis - currentRequestTimeTaken`.
In any case, I would tackle this in a different PR as the rest of the changes are more straightforward.

Reply:
ok i will address this in other pr